### DISCUSSION TITLE
WalletConnectCsharp uses Portable.BouncyCastle package instead of official BouncyCastle.Cryptography build.

### DISCUSSION THREAD

**Discussion by bugbytesinc** [WalletConnect/walletconnect-monorepo#3863] [Developer Support] [answered]
I was wondering, when I look at the package references for the [WalletConnectSharp.Crypto](https://github.com/WalletConnect/WalletConnectSharp/blob/2.0/Core%20Modules/WalletConnectSharp.Crypto/WalletConnectSharp.Crypto.csproj) project, I see a reference to `Portable.BouncyCastle`.  If I go to the Bouncy Castle [website](https://www.bouncycastle.org/csharp/) it suggests that [BouncyCastle.Cryptography](https://www.nuget.org/packages/BouncyCastle.Cryptography/2.2.1) is the official build.  Is the `Portable.BouncyCastle` considered a better build for this library?

**Comment by skibitsky** [comment, marked as answer]
We opted for `Portable.BouncyCastle` because it works better with Unity. On NuGet, it's been downloaded 34 times more than `BouncyCastle.Cryptography`, making it a preferred choice by many devs.

**Reply by bugbytesinc** [reply]
Do you see a practical difference between the two?  The reason I ask is I had a downstream conflict in versions of the BC library. I tried using W.C.Sharp & another lib that pulled in yet a different BouncyCastle nuget package.  The integration tests failed at run time because the other version of BC was loaded before the one used by WC which was missing a constructor for Ed25519KeyParam that WC Core depends on.  I should of known this could have happened due to how the assembly loader works, but it caught me by surprise.  I have control over the former library and could switch it to Portable, I'm just wondering if there are any practical or performance issues.  (being able to support unity would be an extra bonus)
